test/deps/executables_test.dart

```diff
@@ -0,0 +1,245 @@
+// Copyright (c) 2016, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:test/test.dart';
+
+import '../descriptor.dart' as d;
+import '../test_pub.dart';
+
+const _validMain = 'main() {}';
+
+main() {
```

Comment:
I haven't found a good way to test the same things.

I was choosing between:

* for loop
* current approach
* even more test-fn-generators